Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -390,594 +169,217 @@ CGImageRef CGImageMaskCreate(size_t width,
                              CGDataProviderRef provider,
                              const CGFloat* decode,
                              bool shouldInterpolate) {
-    FAIL_FAST_HR_IF_FALSE(E_UNEXPECTED, ((bitsPerComponent == 8) && (bitsPerPixel == 32)));
-    FAIL_FAST_HR_IF_FALSE_MSG(E_INVALIDARG,
-                              ((provider == nullptr) || [(NSObject*)provider isKindOfClass:[NSData class]]),
-                              "CGDataProviderRef does not derive from NSData!");
+    // Native platform support
```

Comment:
>Native platform suppo [](start = 7, length = 21)

could you elaborate this comment?  We are going from fail fast to return NULL, why?